### DISCUSSION TITLE
Fading in switch

### DISCUSSION THREAD

**Discussion by gammaw** [savonet/liquidsoap#2127] [Q&A] [answered]
I try to spawn a new issue from https://github.com/savonet/liquidsoap/issues/1990 which had been originally added as a syntax error related issue. 

SWITCH BASED OPTION (A)
So I'd like to have something like this. The scripts runs but I hear no fading at all: 

```
def fading_transition(a,b)
	sequence(merge=true, [fade.out(a),playlist_jingle_station_id,fade.in(b)])
end

radio = switch(track_sensitive=false, 
    transitions=[fading_transition,fading_transition],
    [({3w and 15h43m-15h44m or 15h45m-15h46m}, delay(60., playlist_my_show)),
    ({true}, playlist_off_air) ])
```

CROSS BASED OPTION (B)
As a workaround (inspired by https://github.com/savonet/liquidsoap/discussions/2123) this one seems to work: 

```
radio = switch(track_sensitive=false, 
    [({3w and 15h43m-15h44m or 15h45m-15h46m}, delay(60., playlist_my_show)),
    ({true}, playlist_off_air) ])

def fading_transition(a,b)
	sequence(merge=true, [fade.out(a.source),playlist_jingle_station_id,fade.in(b.source)])
end

radio = cross(fading_transition, radio)
```

I'd prefer option A though because there I can easily define different transition functions depending on the source leaving/entering. Any idea why option A does not work (and why there is a difference in syntax regarding `a` and `a.source`)?

**Comment by toots** [comment, marked as answer]
Transitions in switches are, unfortunately, more limited than in crossfade. switches do not buffer data so, when the transition is triggered, there is usually very little data available on the ending track and, so the fade out does not work. We've discussing removing switch-based transition because these limitations make them deceiving.

**Reply by gammaw** [reply]
Thanks! Now that I know, I won't try with those transitions (in a switch). I guess I will be able to work it around using suitable metadata in files so that I can decide which exact sources are active in the transition (in our case mostly playlists and live sources).
